cpp/src/arrow/flight/flight_benchmark.cc

```diff
@@ -67,7 +73,113 @@ struct PerformanceStats {
   }
 };
 
-Status RunPerformanceTest(const std::string& hostname, const int port) {
+Status WaitForReady(FlightClient* client) {
+  Action action{"ping", nullptr};
+  for (int attempt = 0; attempt < 10; attempt++) {
+    std::unique_ptr<ResultStream> stream;
+    if (client->DoAction(action, &stream).ok()) {
+      return Status::OK();
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+  }
+  return Status::IOError("Server was not available after 10 attempts");
+}
+
+arrow::Result<PerformanceResult> RunDoGetTest(FlightClient* client,
+                                              const perf::Token& token,
+                                              const FlightEndpoint& endpoint) {
+  std::unique_ptr<FlightStreamReader> reader;
+  RETURN_NOT_OK(client->DoGet(endpoint.ticket, &reader));
+
+  FlightStreamChunk batch;
+
+  // This is hard-coded for right now, 4 columns each with int64
+  const int bytes_per_record = 32;
+
+  // This must also be set in perf_server.cc
+  const bool verify = false;
+
+  int64_t num_bytes = 0;
+  int64_t num_records = 0;
+  while (true) {
+    RETURN_NOT_OK(reader->Next(&batch));
+    if (!batch.data) {
+      break;
+    }
+
+    if (verify) {
+      auto values = reinterpret_cast<const int64_t*>(
+          batch.data->column_data(0)->buffers[1]->data());
```

Comment:
I think you can write `auto values = batch.data->column_data(0)->GetValues<int64_t>(1)`.